cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexManager.java

```diff
@@ -71,80 +74,24 @@
-    public static final String INDEX_METADATA_TABLE_NAME = "_t_cloudant_sync_query_metadata";
-
-    private static final String EXTENSION_NAME = "com.cloudant.sync.query";
-    private static final String INDEX_FIELD_NAME_PATTERN = "^[a-zA-Z][a-zA-Z0-9_]*$";
-
     private static final Logger logger = Logger.getLogger(IndexManager.class.getName());
 
     private final Datastore datastore;
-    private final SQLDatabase database;
-    private final Pattern validFieldName;
-    private final ExecutorService queue;
-
-    private boolean textSearchEnabled;
 
     /**
      *  Constructs a new IndexManager which indexes documents in 'datastore'
      *  @param datastore The {@link Datastore} to index
      */
     public IndexManager(Datastore datastore) {
         this.datastore = datastore;
-        validFieldName = Pattern.compile(INDEX_FIELD_NAME_PATTERN);
-        queue = Executors.newSingleThreadExecutor();
-
-        final String filename = ((DatastoreImpl)datastore).extensionDataFolder(EXTENSION_NAME) + File.separator
-                                                                              + "indexes.sqlite";
-        final KeyProvider keyProvider = ((DatastoreImpl)datastore).getKeyProvider();
 
-        SQLDatabase sqlDatabase = null;
-        try {
-            sqlDatabase = queue.submit(new Callable<SQLDatabase>() {
-                @Override
-                public SQLDatabase call() throws Exception {
-                    SQLDatabase db = SQLDatabaseFactory.openSqlDatabase(filename, keyProvider);
-
-                    if (db != null) {
-                        SQLDatabaseFactory.updateSchema(db,
-                                new SchemaOnlyMigration(QueryConstants.getSchemaVersion1()), 1);
-                        SQLDatabaseFactory.updateSchema(db,
-                                new SchemaOnlyMigration(QueryConstants.getSchemaVersion2()), 2);
-                    }
-
-                    return db;
-                }
-            }).get();
-        } catch (InterruptedException e) {
-            logger.log(Level.SEVERE, "Problem opening or creating database.", e);
-        } catch (ExecutionException e) {
-            logger.log(Level.SEVERE, "Problem opening or creating database.", e);
-        }
-        database = sqlDatabase;
-        textSearchEnabled = ftsAvailable(queue, database);
     }
 
     public void close() {
-        try {
-            queue.submit(new Runnable() {
-                @Override
-                public void run() {
-                    database.close();
-
-                }
-            }).get();
-        } catch (InterruptedException e) {
-            logger.log(Level.SEVERE,"Failed to close db",e);
-            throw new RuntimeException(e);
-        } catch (ExecutionException e) {
-            logger.log(Level.SEVERE, "Failed to close db", e);
-        }
-        queue.shutdown();
+        // do nothing.
```

Comment:
I think it would help anyone debugging into this method to understand _why_ it does nothing. Please add some additional explanation here - something along the lines of "The index lifecycle is now combined with the datastore lifecycle so this deprecated method need not close anything as the index will be shutdown when the datastore is closed."